### PR TITLE
DAOS-7590 test: use container labels with daos by default

### PR DESCRIPTION
**Pull request by daltonbohning**
Test-tag: pr container datamover test_nvme_fragmentation test_nvme_io_verification test_nvme_server_restart

- Adjust LabelGenerator to store a unique counter for each label
- Add TestContainer.label_generator to generate a unique label
  - Currently only when a label value is already supplied
- Add TestContainer.identifier to return the label if defined
- Add TestContainer.use_label to tell the framework to use labels by default
- Adjust TestContainer functions to use self.identifier instead of self.uuid
- Simplify some control_method checks in TestContainer

Required-githooks: true

Signed-off-by: Dalton Bohning <dalton.bohning@intel.com>

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [x] You are the appropriate gatekeeper to be landing the patch.
* [x] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [x] Githooks were used. If not, request that user install them and check copyright dates.
* [x] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [x] All builds have passed.  Check non-required builds for any new compiler warnings.
* [x] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [x] If applicable, the PR has addressed any potential version compatibility issues.
* [x] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [x] Extra checks if forced landing is requested
  * [x] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [x] No new NLT or valgrind warnings.  Check the classic view.
  * [x] Quick-build or Quick-functional is not used.
* [x] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/ftest/util/apricot/apricot/test.py [hunk 1]
```diff
@@ -639,7 +639,8 @@ def __init__(self, *args, **kwargs):
         self.job_manager = None
         # whether engines ULT stacks have been already dumped
         self.dumped_engines_stacks = False
-        self.label_generator = LabelGenerator()
+        self.pool_label_generator = LabelGenerator()
+        self.cont_label_generator = LabelGenerator()
```
Comment (phender) [inline src/tests/ftest/util/apricot/apricot/test.py:643]: Maybe we should use a single label generator object that could maintain unique counts per base label, e.g. A1, A2, B1, A3, B2, A4, etc.

**Review by phender — CHANGES_REQUESTED** [review]

File: src/tests/ftest/util/apricot/apricot/test.py (hunk 1 above)
Comment (daltonbohning) [reply src/tests/ftest/util/apricot/apricot/test.py:643]: Yes, agree. Thanks

**Review by shimizukko — APPROVED** [review]

File: src/tests/ftest/util/apricot/apricot/test.py (hunk 1 above)
Comment (daltonbohning) [reply src/tests/ftest/util/apricot/apricot/test.py]: Addressed in the latest merge commit

**Comment by daltonbohning** [comment]
Existing failures in https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-11324/5/tests
- daos_test/suite.py - https://daosio.atlassian.net/browse/DAOS-12564

**Review by phender — APPROVED** [review]

**Review by shimizukko — APPROVED** [review]

**Pull request merged**